install-ama-arc-extension/terraform/main.tf

```diff
@@ -0,0 +1,152 @@
+provider "azurerm" {
+  features {}
+  skip_provider_registration = "true"
+}
+
+terraform {
+  required_providers {
+    azapi = {
+      source = "azure/azapi"
+    }
+  }
+}
+
+provider "azapi" {
+}
+
+
+# Variables
+variable "clusterResourceIds" {
+  type        = list(string)
+  description = "Resource Ids of the Azure Stack HCI clusters to install the Azure Monitoring Agent extension."
+}
+
+variable "location" {
+  type        = string
+  description = "Location where the data collection rule is to be created."
+}
+
+variable "workspaceId" {
+  type        = string
+  description = "Workspace Id of the workspace to be associated with AMA agent extension."
+}
+
+variable "workspaceName" {
+  type        = string
+  description = "Name of the Workspace to be associated with AMA agent extension."
+}
+
+variable "resourceGroupName" {
+  type        = string
+  description = "Name of the resource group where the log analytics workspace exists."
+}
+
+
+# locals
+locals {
+  AMAExtensionName       = "AzureMonitorWindowsAgent"
+  publisher              = "Microsoft.Azure.Monitor"
+  dataCollectionRuleName = "${var.resourceGroupName}-DataCollectionRule"
+}
+
+
+# Existing resource group where log analytics workspace exists
+data "azurerm_resource_group" "resourceGroup" {
+  name = var.resourceGroupName
+}
+
+# Existing Azure Stack HCI Clusters
+data "azapi_resource" "cluster" {
+  for_each    = toset(var.clusterResourceIds)
+  type        = "Microsoft.AzureStackHCI/clusters@2022-12-01"
```

Comment:
Can we take API version also as variable

Reply:
Changes done as requested.